runtime/runtime.m

```diff
@@ -2118,6 +2118,16 @@ -(void) xamarinSetGCHandle: (int) gc_handle;
 	return res;
 }
 
+void
+xamarin_release_block_on_main_thread (void *obj)
+{
+	if ([NSThread isMainThread]) {
+		_Block_release (obj);
+	} else {
+		dispatch_async_f (dispatch_get_main_queue (), obj, (dispatch_function_t) _Block_release);
```

Comment:
which breaks here 💥 
:(

Reply:
Nice :) I wondered how you could get the value before then.  

note: worth a comment in the source (cuz I'll forget)

My original idea was to avoid ObjC with something like
```
dispatch_queue_t main = dispatch_get_main_queue ();
if (dispatch_get_current_queue () == main) {
    _Block_release (obj);
} else {
    dispatch_async_f (main, obj, (dispatch_function_t) _Block_release);
}
```
but `dispatch_get_current_queue ()` being deprecated I guess it makes this code require a mix of ObjC and C/GDC code.

Reply:
Yes, I agree it looks strange, but all google could come up with about how to check for the main thread was `[NSThread isMainThread]`.

There's ObjC API to schedule things on the main thread, but it seemed (I didn't actually test it though) that it would use more memory (for instance `[NSObject performSelectorOnMainThread:withObject:waitUntilDone:]`, but it requires an NSObject instance). `NSRunLoop` isn't thread-safe so you end up with a chicken-and-egg problem.